public/static/js/version.js

```diff
@@ -39,6 +68,47 @@ mciModule.controller('VersionController', function($scope, $rootScope, $location
     }, 0)
   }
 
+  $scope.addSubscription = function() {
+    promise = addSubscriber($mdDialog, $scope.triggers);
+
+    $mdDialog.show(promise).then(function(data){
+      addSelectorsAndOwnerType(data);
+      $scope.subscriptions.push(data);
+      $scope.saveSubscriptions();
+    });
+  };
+
+  addSelectorsAndOwnerType = function(subscription) {
```

Comment:
Can we dedupe the functions with this name by parameterising  the id and object type? 

Also, I'd prefer to see a function that returns the selectors given parameters, instead of modifying the input parameters

Reply:
Moved to a common function. Since the incomplete subscription object isn't used at all, it felt weird making a copy of it to return so I left that part as is, but let me know if you feel strongly about it